cluster-state-service/internal/features/wrappers/iam_wrapper.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2016-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package wrappers
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/pkg/errors"
+)
+
+type IAMWrapper struct {
+	client *iam.IAM
+}
+
+func NewIAMWrapper() IAMWrapper {
+	awsSession := newAWSIAMSession()
+	return IAMWrapper{
+		client: iam.New(awsSession),
+	}
+}
+
+func newAWSIAMSession() *session.Session {
```

Comment:
This piece of code seems duplicated across various aws sessions we have except for the endpoint piece. There may be value in factoring it out.